composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/ui/components/HyperlinkText.kt

```diff
@@ -0,0 +1,43 @@
+package sp.bvantur.tasky.core.ui.components
+
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.withStyle
+
+@Composable
+fun HyperlinkText(modifier: Modifier = Modifier, allText: String, hyperlinkText: String, onClick: () -> Unit) {
+    val annotation = "clickable"
+
+    val annotatedString = buildAnnotatedString {
+        val hyperLinkStartIndex = allText.indexOf(hyperlinkText)
+
+        append(allText.substring(0, hyperLinkStartIndex).uppercase())
+
+        pushStringAnnotation(tag = annotation, annotation = annotation)
+        withStyle(style = SpanStyle(color = MaterialTheme.colorScheme.secondary)) {
+            append(hyperlinkText.uppercase())
+        }
+        pop()
+
+        append(allText.substring(hyperLinkStartIndex + hyperlinkText.length, allText.length).uppercase())
+    }
+
+    ClickableText(
+        modifier = modifier,
+        text = annotatedString,
+        onClick = { offset ->
+            annotatedString.getStringAnnotations(
+                tag = annotation,
+                start = offset,
+                end = offset
+            ).firstOrNull()?.let {
+                onClick()
+            }
+        },
+        style = MaterialTheme.typography.bodyMedium
+    )
```

Comment:
I personally found that with the latest Compose version creating annotated strings is a bit easier by converting them from HTML (it's debatable, if that should be done generally, but in case you didn't know):
```
Text(
    text = AnnotatedString
        .fromHtml(
            htmlString = """
                <a href="https://pl-coding.com">
                    This is my clickable link
                </a>
                """.trimIndent(),
            linkStyles = TextLinkStyles(
                style = SpanStyle(
                    color = Color.Blue
                )
            ),
            linkInteractionListener = { 
                println("Clicked link")
            }
        ),
)
```

Reply:
Nice to know that. I will keep the one that I currently have since I spent some time on it. I will also store your approach and try it out just to see how does it works.

Reply:
Actually, I think this one is not available yet for Compose Multiplatform. In documentation, there is only an Android icon next to this extension method so I guess we will need to wait for support on Compose Multiplatform for this. 😅 

https://developer.android.com/reference/kotlin/androidx/compose/ui/text/AnnotatedString.Companion

Reply:
Ah right, CMP is a bit behind 😅 Then you'll at least know of it for the future 👍